<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>636</width>
    <height>371</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>320</width>
    <height>240</height>
   </size>
  </property>
  <property name="acceptDrops">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>Bullet Physics Playground - (loading..)</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../res.qrc">
    <normaloff>:/icons/bpp.svg</normaloff>:/icons/bpp.svg</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="0">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="Viewer" name="viewer" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>320</width>
        <height>240</height>
       </size>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>636</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionReload"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionPreferences"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="action_Homepage"/>
    <addaction name="separator"/>
    <addaction name="action_Chat"/>
    <addaction name="action_Issues"/>
    <addaction name="action_Wiki"/>
    <addaction name="separator"/>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menu_View">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <addaction name="action_Full_screen"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menu_View"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBarFile">
   <property name="windowTitle">
    <string>File</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionReload"/>
   <addaction name="actionSave"/>
   <addaction name="actionSaveAs"/>
  </widget>
  <widget class="QToolBar" name="toolBarEdit">
   <property name="windowTitle">
    <string>Edit</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionPreferences"/>
  </widget>
  <widget class="QToolBar" name="toolBarSim">
   <property name="windowTitle">
    <string>Simulation</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionToggleSim"/>
   <addaction name="actionRestart"/>
   <addaction name="separator"/>
   <addaction name="actionToggleDeactivation"/>
   <addaction name="actionTogglePOVExport"/>
  </widget>
  <widget class="QToolBar" name="toolBarView">
   <property name="windowTitle">
    <string>View</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionResetCamera"/>
   <addaction name="separator"/>
  </widget>
  <widget class="QToolBar" name="Window">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_Full_screen"/>
  </widget>
  <action name="actionNew">
   <property name="icon">
    <iconset theme="document-new">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;New</string>
   </property>
   <property name="toolTip">
    <string>Create a new file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset theme="document-open">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Open..</string>
   </property>
   <property name="toolTip">
    <string>Open an existing file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset theme="document-save">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Save..</string>
   </property>
   <property name="toolTip">
    <string>Save file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="icon">
    <iconset theme="document-save-as">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Save &amp;As..</string>
   </property>
   <property name="toolTip">
    <string>Save file as</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset theme="exit">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>E&amp;xit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionPreferences">
   <property name="icon">
    <iconset theme="preferences-system">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Preferences..</string>
   </property>
   <property name="toolTip">
    <string>Edit application preferences</string>
   </property>
   <property name="shortcut">
    <string>F12</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>&amp;About</string>
   </property>
   <property name="toolTip">
    <string>Show the application's About box</string>
   </property>
  </action>
  <action name="actionToggleSim">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="media-playback-start">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Start/Stop simulation</string>
   </property>
   <property name="toolTip">
    <string>Run/Pause simulation</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionRestart">
   <property name="icon">
    <iconset theme="view-refresh">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Restart</string>
   </property>
   <property name="toolTip">
    <string>Restart simulation</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionTogglePOVExport">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../res.qrc">
     <normaloff>:/icons/povray.svg</normaloff>:/icons/povray.svg</iconset>
   </property>
   <property name="text">
    <string>Toggle POV Export</string>
   </property>
  </action>
  <action name="actionToggleDeactivation">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../res.qrc">
     <normaloff>:/icons/deactivation.png</normaloff>:/icons/deactivation.png</iconset>
   </property>
   <property name="text">
    <string>Toggle deactivation</string>
   </property>
   <property name="toolTip">
    <string>Enable / disable automatic state deactivation feature</string>
   </property>
  </action>
  <action name="actionResetCamera">
   <property name="icon">
    <iconset theme="go-home">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Reset camera view to initial state</string>
   </property>
  </action>
  <action name="action_Issues">
   <property name="text">
    <string>&amp;Issues</string>
   </property>
   <property name="toolTip">
    <string>Bullet Physiscs Playground issues on GitHub</string>
   </property>
  </action>
  <action name="action_Wiki">
   <property name="text">
    <string>&amp;Wiki</string>
   </property>
  </action>
  <action name="action_Homepage">
   <property name="text">
    <string>&amp;Homepage</string>
   </property>
   <property name="toolTip">
    <string>http//bullet-physics-playground.github.io/</string>
   </property>
  </action>
  <action name="action_Chat">
   <property name="text">
    <string>&amp;Chat</string>
   </property>
  </action>
  <action name="action_RenderFrame">
   <property name="text">
    <string>&amp;Quick Render</string>
   </property>
   <property name="toolTip">
    <string>Render current frame with POV-Ray</string>
   </property>
   <property name="shortcut">
    <string>F6</string>
   </property>
  </action>
  <action name="actionQuickRender">
   <property name="icon">
    <iconset theme="camera-photo">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Quick Render</string>
   </property>
   <property name="toolTip">
    <string>Render current frame with POV-Ray</string>
   </property>
   <property name="shortcut">
    <string>F6</string>
   </property>
  </action>
  <action name="action_Full_screen">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="window_fullscreen">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>&amp;Full screen</string>
   </property>
   <property name="shortcut">
    <string>F11</string>
   </property>
  </action>
  <action name="actionReload">
   <property name="icon">
    <iconset theme="document-revert">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Reload file</string>
   </property>
   <property name="toolTip">
    <string>Reload currently open file from disk</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Viewer</class>
   <extends>QWidget</extends>
   <header>viewer.h</header>
   <container>1</container>
   <slots>
    <slot>onQuickRender(QString)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../res.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionNew</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileNew()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>321</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>321</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>321</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSaveAs</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileSaveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>321</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPreferences</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>editPreferences()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>321</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>321</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>helpAbout()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>321</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionToggleSim</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleSim()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>321</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRestart</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>rerunProgram()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>321</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionTogglePOVExport</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>togglePOVExport(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>321</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionToggleDeactivation</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleDeactivation(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>321</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionResetCamera</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>resetCamera()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>321</x>
     <y>166</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Homepage</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>helpHomepage()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>224</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Issues</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>helpIssues()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>224</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Wiki</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>helpWiki()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>224</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Chat</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>helpChat()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>224</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Full_screen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>toggleFullscreen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>317</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionReload</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>fileReload()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>317</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>fileNew()</slot>
  <slot>fileOpen()</slot>
  <slot>fileReload()</slot>
  <slot>fileSave()</slot>
  <slot>fileSaveAs()</slot>
  <slot>fileExit()</slot>
  <slot>editPreferences()</slot>
  <slot>helpAbout()</slot>
  <slot>rerunProgram()</slot>
  <slot>toggleSim()</slot>
  <slot>togglePOVExport(bool)</slot>
  <slot>toggleScreenshotExport(bool)</slot>
  <slot>toggleDeactivation(bool)</slot>
  <slot>resetCamera()</slot>
  <slot>helpHomepage()</slot>
  <slot>helpIssues()</slot>
  <slot>helpWiki()</slot>
  <slot>helpChat()</slot>
  <slot>toggleFullscreen()</slot>
 </slots>
</ui>
